lib/routes/account.js

```diff
@@ -1257,49 +1257,116 @@ module.exports = function (
         var authPW = Buffer(request.payload.authPW, 'hex')
         var authSalt = crypto.randomBytes(32)
         var password = new Password(authPW, authSalt, config.verifierVersion)
-        return password.verifyHash()
-          .then(
-            function (verifyHash) {
-              return db.resetAccount(
-                accountResetToken,
-                {
-                  authSalt: authSalt,
-                  verifyHash: verifyHash,
-                  wrapWrapKb: crypto.randomBytes(32),
-                  verifierVersion: password.version
-                }
-              )
-            }
-          )
-          .then(
-            function () {
-              // Notify all devices that the account has changed.
-              push.notifyUpdate(accountResetToken.uid, 'passwordReset')
+        var account, sessionToken, keyFetchToken, verifyHash, wrapKb
 
-              return db.account(accountResetToken.uid)
-            }
-          )
-          .then(
-            function (account) {
-              log.activityEvent('account.reset', request, {
-                uid: account.uid.toString('hex')
+        return verifyAndResetAccount()
+          .then(createSessionToken)
```

Comment:
As @rfk notes below, this function now *always* creates a new sessionToken, whereas the keyFetchToken is only created if `keys=true`. Can sessionToken creation be opt-in as well. I'm someone indifferent to what signal should indicate "create a sessionToken", whether we use `keys=true` or rfk's suggestion of `sessionToken: true` in the request body. If the latter, keyFetchToken's would only be created if both `sessionToken: true` and `?keys=true` were specified.